core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -749,6 +752,43 @@ class DAGSchedulerSuite
     assertDataStructuresEmpty()
   }
 
+  /**
+   * Makes sure that tasks for a stage used by multiple jobs are submitted with the properties of a
+   * later, active job if they were previously run under a job that is no longer active
+   */
+  test("stage used by two jobs, the first no longer active") {
+    val baseRdd = new MyRDD(sc, 1, Nil)
+    val finalRdd1 = new MyRDD(sc, 1, List(new OneToOneDependency(baseRdd)))
+    val finalRdd2 = new MyRDD(sc, 1, List(new OneToOneDependency(baseRdd)))
```

Comment:
is the problem that w/ a OnetoOneDependency, there isn't actually a shared stage?  I think there will be a shared RDD,  but the stages are still separate.  I'm thinking you need a test where you have a shared ShuffleDependency.